### DISCUSSION TITLE
[Feature] Keep track of previously deleted files

### DISCUSSION THREAD

**Discussion by mrevl** [immich-app/immich#13365] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

It would be great if Immich could keep track of files that were explicitly deleted and can re-tag those as "previously deleted" when they are imported again.

E.g. You import a whole bunch of photo's and part of those pictures were already imported and part of them were already deleted. The duplicate pictures are obvious, but the pictures previously removed, are not. This feature would solve that.

Possible solution - keep track of the picture metadata that's deleted, when a new picture arrives, check if it got deleted previously and if so, add a deleted tag inside the picture database.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by niranjanshr13** [comment]
no.

**Comment by bo0tzz** [comment]
This is basically what the trash is for. In the admin settings you can change how long files stay in the trash before automatically being deleted, if that'd help.

**Reply by Acidiphilium** [reply]
Not quite same. Trash takes space in disk, it's why there's a time limit. After that time, there is no way to know if a reimported file have already been deleted once. This small feature/request is not useless and maybe could be done like a little flag or tag after keeping the image/movie hash into the database so can then be easy to flag it as deleted when reimported. I'm in the same situation as mrevl. I'm doing right now many imports of old backups into immich and some are from duplicated folders/images and other people who could have same pictures which I may have already deleted. Keeping the trash full forever is not a nice solution for that little problem.
